lib/glific/saas/queries.ex

```diff
@@ -48,8 +53,47 @@ defmodule Glific.Saas.Queries do
     result
     |> validate_bsp_keys(params)
     |> validate_shortcode(params["shortcode"])
-    |> validate_email(params["email"])
     |> validate_phone(params["phone"])
+    |> validate_text_field(params["name"], :name, {1, 40})
+  end
+
+  @doc """
+  Validate all the details regarding NGO registration
+  """
+  @spec validate_registration_details(map(), map()) :: map()
+  def validate_registration_details(result, params) do
+    Enum.reduce(params, result, fn {key, value}, result ->
+      case key do
+        "billing_frequency" ->
+          validate_billing_frequency(result, value)
+
+        "finance_poc" ->
+          validate_finance_poc(result, value)
+
+        "submitter" ->
+          validate_submitter_details(result, value)
+
+        "signing_authority" ->
+          validate_signer_details(result, value)
+
+        "org_details" ->
+          validate_org_details(result, value)
+
+        _ ->
+          result
```

Comment:
is this tested with multiple errors and more than one  function returning error? The error I don't see to been accumulated